Enhancements on UX flows section (#1328)
vmware/versatile-data-kit: specs/vep-994-jupyter-notebook-integration/README.md

|       Create job       |    After the user enters JupyterLab he will see the VDK drop down menu where he can find the "Create job" option. After clicking that option the user get's a pop up where he should enter the needed information for the job that will be created: name, team, directory and whether it will only be created locally or in the cloud as well. After filling all the needed information for the job by clicking a "Create" button it will be created.     |         1          |   high   |
|       Delete job       |                                                                                                                                                                                                  "Delete job" will be introduced just like "Create job".                                                                                                                                                                                                  |         1          |   mid    |

> |  Work with SQL steps   |                                                                   One SQL step can be introduced as one cell. The cells that will be part of the job should have a tag ("vdk", "vdk-sql") that shows that the cell should be included in the job. All the cells that does not have that tag will not be included in he job as a step. One notebook file can have as several SQL steps.                                                                    |      2,3,4,7       |   high   |

Reviewer: ```suggestion
|  Work with SQL steps   |                                                                   One SQL step can be introduced as one cell. The cells that will be part of the job should have a tag ("vdk", "vdk-sql") that shows that the cell should be included in the job. All the cells that do not have that tag will not be included in the job as a step. One notebook file can have several SQL steps.                                                                    |      2,3,4,7       |   high   |
```

| Error preventions                                                                                         | 1                                                                                                                                    | 1                                                                                                                                                   |
| Ease of learning: how fast a user who has never seen the user interface before can accomplish basic tasks | 1 <br /> it is harder since the magic commands and writing a function to multiple cells is Jupyter specific                          | 2 <br />  calling functions from one function is a thing we do generally in coding                                                                  |

    Solution 2 total score: 18       Solution 1 total score: 23

> Final decision: Solution 1

Reviewer 2: Mention that both solutions are not Mutually Exclusive. That is at some point we can implement both. But for the first iteration/version based on above analysis we will go with Solution 1.

![example-use-case](example.png)

### Availability
The availability of the extension will be managed by JupyterLab since it is going to be run as part of the JupyterLab ecosystem.

### Test Plan
* The front-end extension is using Jest for JavaScript code testing.
* This extension uses Playwright for the integration tests (aka user level tests). More precisely, the JupyterLab helper Galata is used to handle testing the extension in JupyterLab.
* The server extension is using Pytest for Python code testing.
* The VDK notebook plugin will be using Pytest for Python code testing.

### Security and Permissions

> In terms of security, Jupyter uses tornado to ensure only authorized user can request the Jupyter server

Reviewer 2: You can add a todo only but a few more security concerns need to be looked into 
* What would be the authenticaiton/authorizaiton against the control service  -you can refer to control service doc for this.

> You can read more [here](https://jupyterlab-server.readthedocs.io/en/stable/api/index.html).

Reviewer 2: Making sure to supply references is very important. But in this case I think https://jupyter-notebook.readthedocs.io/en/stable/security.html# is more precise reference for this section.
